python/metatensor-operations/metatensor/operations/_dispatch.py

```diff
@@ -610,3 +610,95 @@ def where(array):
         return np.where(array)
     else:
         raise TypeError(UNKNOWN_ARRAY_TYPE)
+
+
+def take(array, indices, axis: int):
+    """
+    See torch take_along_dim or numpy take as reference.
```

Comment:
```suggestion
    See :py:func:`torch.take_along_dim` or :py:func:`numpy.take` as reference.
```

Reply:
you mention take_along_dim, but then use index_select. Any reason why?

Reply:
ah thanks for the catch, forgot to change the doc, `take_along_dim` did not work with TorchScript (had some assumptions on the shape of input and indices)